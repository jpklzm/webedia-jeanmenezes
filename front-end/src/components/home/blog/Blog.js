import React, { Component } from 'react';

import { get } from '../../../services/post';
import { FeaturedPost, Post } from './post/Post';

class Blog extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      posts: [],
      shouldLoadMorePosts: true,
    };

    this.getPosts = this.getPosts.bind(this);
  }

  componentDidMount() {
    this.getPosts();
  }

  getPosts() {
    get(1).then(response => {
      this.setState({
        posts: this.state.posts.concat(response.data),
        loading: false,
      });
    });
  }

  render() {
    return (
      <section className="container" style={{ width: '80%' }}>
        <article className="row">
          {this.state.posts[0] ? <FeaturedPost post={this.state.posts[0]} /> : null}
        </article>
        {window.innerWidth >= 880 // Desktop view with two columns of small posts
          ? <div className="row">
              {this.state.posts.filter(p => p !== this.state.posts[0]).map((post, key) => (
                <article key={key}>
                  <Post post={post} />
                </article>
              ))}
            </div>
          : // Mobile view with single column and fullsized posts
            <div className="row">
              {this.state.posts.filter(p => p !== this.state.posts[0]).map((post, key) => (
                <article key={key}>
                  <FeaturedPost post={post} />
                </article>
              ))}
            </div>}
      </section>
    );
  }
}

export default Blog;
