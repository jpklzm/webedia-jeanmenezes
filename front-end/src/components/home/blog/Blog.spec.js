import React from 'react';
import renderer from 'react-test-renderer';

import Blog from './Blog';

test('Snapshot', () => {
  const component = renderer.create(<Blog />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
