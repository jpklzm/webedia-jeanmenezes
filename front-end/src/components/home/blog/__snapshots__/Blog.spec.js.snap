// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<section
  className="container"
  style={
    Object {
      "width": "80%",
    }
  }
>
  <article
    className="row"
  />
  <div
    className="row"
  />
</section>
`;
